server/mailet/mailets/src/main/java/org/apache/james/transport/mailets/delivery/MailDispatcher.java

```diff
@@ -141,6 +151,13 @@ public void dispatch(Mail mail) throws MessagingException {
         return errors;
     }
 
+    private Mono<Void> storeMailWithRetry(Mail mail, MailAddress recipient) throws MessagingException {
+       return Mono.fromRunnable((ThrowingRunnable)() -> mailStore.storeMail(recipient, mail))
```

Comment:
Mono.fromCallable() and you can avoid casting

Reply:
.onError(throwable -> logger.) ? Otherwise, the method will return a Mono.error() of retries exhausted exception without the details of the real error

Reply:
> Mono.fromCallable() and you can avoid casting

it throw en exception so i will have some ceremony too.